plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/persistence/mappers/ToDatabaseAddonsMapper.kt

```diff
@@ -0,0 +1,136 @@
+        return when (this) {
+            TitleFormat.Label -> AddonEntity.LocalTitleFormat.Label
+            TitleFormat.Heading -> AddonEntity.LocalTitleFormat.Heading
+            TitleFormat.Hide -> AddonEntity.LocalTitleFormat.Hide
+        }
+    }
+
+    private fun Addon.mapRestrictions(): AddonEntity.LocalRestrictions? {
+        return if (this is Addon.CustomText) {
+            when (this.restrictions) {
+                Restrictions.AnyText -> AddonEntity.LocalRestrictions.AnyText
+                Restrictions.OnlyLetters -> AddonEntity.LocalRestrictions.OnlyLetters
+                Restrictions.OnlyNumbers -> AddonEntity.LocalRestrictions.OnlyNumbers
+                Restrictions.OnlyLettersNumbers -> AddonEntity.LocalRestrictions.OnlyLettersNumbers
+                Restrictions.Email -> AddonEntity.LocalRestrictions.Email
+            }
+        } else {
+            null
+        }
+    }
+
+    private fun Addon.mapPriceType(): AddonEntity.LocalPriceType? {
+        return if (this is HasAdjustablePrice) {
+            val price = this.price
+            if (price is Adjusted) {
+                when (price.priceType) {
+                    PriceType.FlatFee -> AddonEntity.LocalPriceType.FlatFee
+                    PriceType.QuantityBased -> AddonEntity.LocalPriceType.QuantityBased
+                    PriceType.PercentageBased -> AddonEntity.LocalPriceType.PercentageBased
+                }
+            } else {
+                null
+            }
+        } else {
+            null
+        }
+    }
+
+    private fun Addon.mapType(): AddonEntity.LocalType {
+        return when (this) {
+            is Addon.Checkbox -> AddonEntity.LocalType.Checkbox
+            is Addon.CustomPrice -> AddonEntity.LocalType.CustomPrice
+            is Addon.CustomText -> AddonEntity.LocalType.CustomText
+            is Addon.CustomTextArea -> AddonEntity.LocalType.CustomTextArea
+            is Addon.FileUpload -> AddonEntity.LocalType.FileUpload
+            is Addon.Heading -> AddonEntity.LocalType.Heading
+            is Addon.InputMultiplier -> AddonEntity.LocalType.InputMultiplier
+            is Addon.MultipleChoice -> AddonEntity.LocalType.MultipleChoice
+        }
+    }
+
+    private fun Addon.mapDisplay(): AddonEntity.LocalDisplay? {
+        return if (this is Addon.MultipleChoice) {
+            when (this.display) {
+                Display.Select -> AddonEntity.LocalDisplay.Select
+                Display.RadioButton -> AddonEntity.LocalDisplay.RadioButton
+                Display.Images -> AddonEntity.LocalDisplay.Images
+            }
+        } else {
+            null
+        }
+    }
+
+    private fun Addon.mapPrice(): String? {
+        return if (this is HasAdjustablePrice) {
+            val price = this.price
+            if (price is Adjusted) {
+                price.value
+            } else {
+                null
+            }
+        } else {
+            null
+        }
+    }
```

Comment:
np: you can greatly improve this function readability by taking advantage of kotlin null returning operations
```suggestion
    private fun Addon.mapPrice(): String? =
            (this as? HasAdjustablePrice)
                    ?.let { it.price as? Adjusted }
                    ?.value
```

Reply:
love it, much more readable!